src/Nethermind/Nethermind.Network.Discovery/Lifecycle/NodeLifecycleManager.cs

```diff
@@ -190,14 +190,24 @@ private void ProcessNodes(NeighborsMsg msg)
         NodeStats.AddNodeStatsEvent(NodeStatsEventType.DiscoveryNeighboursIn);
         RefreshNodeContactTime();
 
-        foreach (Node node in msg.Nodes)
+        foreach (Node? node in msg.Nodes)
         {
+            if (node is null)
+            {
+                // ports differed between tcp and udp
+                return;
+            }
             if (node.Address.Address == _localhost)
             {
                 if (_logger.IsTrace)
                     _logger.Trace($"Received localhost as node address from: {msg.FarPublicKey}, node: {node}");
                 continue;
             }
+            else if (!NetworkStorage.NodesFilter.Set(node.Address.Address))
```

Comment:
I believe this should be better placed in `DiscoveryManager`.

Reply:
Moved to DiscoveryManager